src/main/java/sg/ncl/MainController.java

```diff
@@ -471,10 +477,117 @@ public String loginSubmit(
 
     }
 
-    @RequestMapping("/passwordreset")
-    public String passwordreset(Model model) {
-        model.addAttribute("loginForm", new LoginForm());
-        return "passwordreset";
+    // triggered when user clicks "Forget Password?"
+    @RequestMapping("/password_reset_email")
+    public String passwordResetEmail(Model model) {
+        model.addAttribute("passwordResetRequestForm", new PasswordResetRequestForm());
+        return "password_reset_email";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "password_reset_email" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)